signer/examples/relay-client.rs

```diff
@@ -0,0 +1,114 @@
+use clap::Parser;
+
+use p256k1::keys::PublicKey;
+use p256k1::scalar::Scalar;
+use rand::rngs::OsRng;
+use signer::ecdsa;
+use signer::message;
+use signer::network::grpc_relay;
+
+use signer::network::MessageTransfer;
+
+type Msg = ecdsa::Signed<message::SignerMessage>;
+
+/// Relay client example for sBTC v1.
+///
+/// This example shows how to connect to a relay server
+/// and send/receive messages from it. In particular,
+/// this binary will send random messages on a regular
+/// interval to the relay server and display any
+/// incoming messages.
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    #[arg(long, default_value_t = String::from("127.0.0.1"), env = "SBTC_RELAY_HOST")]
+    host: String,
+
+    #[arg(long, default_value_t = 50042, env = "SBTC_RELAY_PORT")]
+    port: u16,
+
+    #[arg(long, default_value_t = true, env = "SBTC_LOG_PRETTY")]
+    pretty_logs: bool,
+
+    #[arg(
+        short,
+        long,
+        default_value_t = 2,
+        env = "SBTC_EXAMPLE_RELAY_CLIENT_PERIOD"
+    )]
+    send_period_seconds: u64,
+}
+
+impl Args {
+    fn address(&self) -> String {
+        let Self { host, port, .. } = self;
+        format!("http://{host}:{port}")
+    }
+
+    fn send_period(&self) -> std::time::Duration {
+        std::time::Duration::from_secs(self.send_period_seconds)
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    let args = Args::parse();
+
+    sbtc_common::logging::setup_logging(args.pretty_logs);
+
+    let mut client = grpc_relay::RelayClient::connect(args.address())
+        .await
+        .expect("Failed to connect");
+
+    let private_key = setup_keypair();
+    let mut message_source = spawn_message_source(private_key, args.send_period());
+
+    loop {
+        tokio::select! {
+            msg_result = client.receive() => {
+                match msg_result {
+                    Ok(msg) => tracing::info!(?msg.signer_pub_key, "received message"),
+                    Err(_) => client = grpc_relay::RelayClient::connect(args.address())
+                                .await.expect("Failed to connect"),
+                }
+            }
+
+            Some(msg) = message_source.recv() => {
+                tracing::info!("sent message");
+                client.broadcast(msg).await.expect("Failed to send message");
+            }
+
+            _ = tokio::signal::ctrl_c() => {
```

Comment:
When I tried to run this locally after `cargo build`, it wouldn't find `signal`
It worked after updating the root Cargo.toml to explicitly include the `signal` feature. 
`tokio = { version = "1.32.0", features = ["signal"]  }`
Not sure why.

Reply:
Hmmm, yeah we usually need to specify the "signal" feature. Also weird that I do not have an issue running the relay-client and relay-server. I also I watched @netrome test the binaries on a video call. I'm not sure what is happening here.

Reply:
In either case, I'll explicitly add the feature. Thank you @8marz8 for noticing this! :pray:

Reply:
Okay, so I think I understand to some degree what's going on here now. In `emily/lambda/Cargo.toml` we have `tokio = { workspace = true, features = ["full"] }`, which means that we compile Tokio in the workspace with all the features, and I guess that's reused across all the crates within the workspace (although I don't know where that's specified). While if we build just the signer, we are missing the feature that we need.

Reply:
I've added the feature flag now.